sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -284,7 +284,7 @@ public void clearTask(String taskName) throws StateStoreException {
         try {
             byte[] bytes = curator.get(path);
             if (bytes.length > 0) {
-                return Optional.of(Protos.TaskInfo.parseFrom(bytes));
+                return Optional.of(CommonTaskUtils.unpackTaskInfo(Protos.TaskInfo.parseFrom(bytes)));
```

Comment:
We also use what's in the state store to reconstitute TaskInfos when we're relaunching an existing pod, so there's some tension between the state store's dual role of "record of what we told Mesos" versus "repository for internal domain objects," which our unpacked illegal TaskInfos kind of are by virtue of representing an internal model that Mesos won't accept. The process for packing them and unpacking is deterministic, so I think they still serve their purpose as a debugging tool if you trust the packing/unpacking logic. I guess if the state store never lets them leave its scope in packed form then I'm okay with it.